integration-cli/docker_cli_by_digest_test.go

```diff
@@ -471,8 +470,11 @@ func (s *DockerRegistrySuite) TestPullFailsWithAlteredLayer(c *check.C) {
 		c.Fatalf("unable to decode image manifest from blob: %s", err)
 	}
 
```

Comment:
at the very least check that that layer you removed is the same one on the progress bar line that gets the verification fail text.